tests/components/airly/test_init.py

```diff
@@ -88,13 +90,58 @@ async def test_config_with_turned_off_station(hass, aioclient_mock):
 
 async def test_update_interval(hass, aioclient_mock):
     """Test correct update interval when the number of configured instances changes."""
-    entry = await init_integration(hass, aioclient_mock)
+    headers = {"X-RateLimit-Limit-day": "100", "X-RateLimit-Remaining-day": "15"}
+    point = datetime(
+        year=2030, month=3, day=2, hour=21, minute=20, second=0, tzinfo=NATIVE_UTC
+    )
+
+    with patch("homeassistant.core.dt_util.utcnow") as mock_utcnow:
+        mock_utcnow.return_value = point
```

Comment:
This patch is only applied within the context manager which exits on the next line. Maybe this is the problem?

Reply:
This is an obvious mistake. Unfortunately, the test still fail for dates from the past.

Reply:
We may need to patch the path to the dt util in the integration too?

https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Reply:
The test still fails for dates from the past.

Reply:
Do we need to patch utcnow here too?

https://github.com/home-assistant/core/blob/0f757c3db21c02747668bac89d03715b3fb106f2/homeassistant/helpers/update_coordinator.py#L18

Reply:
I'll check that, thanks.

Reply:
`patch("homeassistant.helpers.update_coordinator.utcnow", return_value=POINT)` and date from the past cause hanging `pytest` (I have to use CTRL+C to stop `pytest`). With date from the future test passes 🤔